smarts/core/sensors.py

```diff
@@ -951,83 +913,71 @@ def teardown(self):
 
 
 class WaypointsSensor(Sensor):
-    def __init__(self, sim, vehicle, mission_planner: MissionPlanner, lookahead=32):
-        self._sim = sim
+    def __init__(self, vehicle, plan: Plan, lookahead=32):
         self._vehicle = vehicle
-        self._mission_planner = mission_planner
+        self._plan = plan
         self._lookahead = lookahead
 
     def __call__(self):
-        if self._mission_planner.mission.task is not None:
-            if isinstance(self._mission_planner.mission.task, UTurn):
-                return self._mission_planner.uturn_waypoints(
-                    self._sim, self._vehicle.pose, self._vehicle
-                )
-            elif isinstance(self._mission_planner.mission.task, CutIn):
-                return self._mission_planner.cut_in_waypoints(
-                    self._sim, self._vehicle.pose, self._vehicle
-                )
-
-        return self._mission_planner.waypoint_paths_at(
-            pose=self._vehicle.pose,
+        return self._plan.road_map.waypoint_paths(
+            self._vehicle.pose,
             lookahead=self._lookahead,
+            route=self._plan.route,
         )
 
     def teardown(self):
         pass
 
 
 class RoadWaypointsSensor(Sensor):
-    def __init__(self, vehicle, sim, mission_planner, horizon=32):
+    def __init__(self, vehicle, sim, plan, horizon=32):
         self._vehicle = vehicle
-        self._road_network = sim.road_network
-        self._mission_planner = mission_planner
+        self._road_map = sim.road_map
+        self._plan = plan
         self._horizon = horizon
 
     def __call__(self):
-        lp = self._road_network.lanepoints.closest_lanepoint(self._vehicle.pose)
-        road_edges = self._road_network.road_edge_data_for_lane_id(lp.lane_id)
-
+        lane = self._road_map.nearest_lane(self._vehicle.pose.point)
+        road = lane.road
         lane_paths = {}
-        for edge in road_edges.forward_edges + road_edges.oncoming_edges:
-            for lane in edge.getLanes():
-                lane_paths[lane.getID()] = self.paths_for_lane(lane)
+        for croad in [road] + road.parallel_roads + road.oncoming_roads:
+            for lane in croad.lanes:
+                lane_paths[lane.lane_id] = self.paths_for_lane(lane)
 
         route_waypoints = self.route_waypoints()
 
         return RoadWaypoints(lanes=lane_paths, route_waypoints=route_waypoints)
 
     def route_waypoints(self):
-        return self._mission_planner.waypoint_paths_at(
-            pose=self._vehicle.pose,
-            lookahead=32,
+        return self._road_map.waypoint_paths(
+            self._vehicle.pose,
+            lookahead=self._horizon,
+            route=self._plan.route,
         )
 
     def paths_for_lane(self, lane, overflow_offset=None):
+        # XXX: the following assumes waypoint spacing is 1m
```

Comment:
Because `self._horizon` is used both as an _offset_ (which implies its units is meters) and as a count of waypoints.  For example, `start_offset = offset - self._horizon` is using it as meters, whereas `wps_to_lookaheaad = self.horizon * 2` is treating it like a waypoint count.